src/frequenz/client/common/pagination/__init__.py

```diff
@@ -75,13 +92,85 @@ def from_proto(cls, pagination_info: PaginationInfo) -> Self:
             next_page_token=pagination_info.next_page_token,
         )
 
-    def to_proto(self) -> PaginationInfo:
+    def to_proto(self) -> PBPaginationInfo:
         """Convert a Info object to protobuf PBPaginationInfo.
 
         Returns:
             Protobuf message corresponding to the Info object.
         """
-        return PaginationInfo(
+        return PBPaginationInfo(
+            total_items=self.total_items,
+            next_page_token=self.next_page_token,
+        )
+
+
+@dataclass(frozen=True, kw_only=True)
+class PaginationParams:
+    """Parameters for paginating list requests."""
+
+    page_size: int
+    """The maximum number of results to be returned per request."""
+
+    page_token: str
+    """The token identifying a specific page of the list results."""
+
+    @classmethod
+    def from_proto(cls, pagination_params: PBPaginationParamsAlpha8) -> Self:
+        """Convert a protobuf PaginationParams to python PaginationParams object.
+
+        Args:
+            pagination_params: Params to convert.
+        Returns:
+            Params object corresponding to the protobuf message.
+        """
+        return cls(
+            page_size=pagination_params.page_size,
+            page_token=pagination_params.page_token,
+        )
+
+    def to_proto(self) -> PBPaginationParamsAlpha8:
+        """Convert a Params object to protobuf PaginationParams.
+
+        Returns:
+            Protobuf message corresponding to the Params object.
+        """
+        return PBPaginationParamsAlpha8(
+            page_size=self.page_size,
+            page_token=self.page_token,
+        )
+
+
+@dataclass(frozen=True, kw_only=True)
+class PaginationInfo:
+    """Information about the pagination of a list request."""
+
+    total_items: int
+    """The total number of items that match the request."""
+
+    next_page_token: str | None = None
+    """The token identifying the next page of results."""
+
+    @classmethod
+    def from_proto(cls, pagination_info: PBPaginationInfoAlpha8) -> Self:
+        """Convert a protobuf PBPaginationInfo to Info object.
+
+        Args:
+            pagination_info: Info to convert.
+        Returns:
+            Info object corresponding to the protobuf message.
+        """
+        return cls(
+            total_items=pagination_info.total_items,
+            next_page_token=pagination_info.next_page_token,
```

Comment:
This also needs fixing (we can fix it only here if we deprecate the old one):
```suggestion
            next_page_token=pagination_info.next_page_token if pagination_info.HasField("next_page_token") else None,
```

Otherwise it will be assigned to `""` when the field is missing.